control-plane/roles/gardener/templates/30-cloud-profile-metal.yaml

```
@@ -30,6 +30,10 @@ spec:
 {% for id, partition_config in gardener_cloud_profile_partitions.items() %}
           {{ id }}:
             firewallTypes: {{ partition_config.get('default-machine-types', {}).get('firewall', []) | to_json }}
+{% if 'network-isolation' in partition_config %}
+            networkIsolation: |
+              {{ partition_config.get('network-isolation') | to_yaml | indent(width=14, first=false) }}
```

Sry, we use in the whole file to_Json.
```suggestion
            networkIsolation: {{ partition_config.get('network-isolation') | to_json }}
```